app/src/main/java/ch/epfl/cs311/wanderwave/model/utils/BeaconGenerator.kt

```diff
@@ -0,0 +1,133 @@
+        "tourist_attraction",
+        "train_station",
+        "transit_station",
+        "university",
+        "zoo")
+
+fun createNearbyBeacons(
+    location: Location,
+    nearbyBeacons: List<Beacon>,
+    radius: Double,
+    context: Context
+): List<Beacon> {
+  // Initialize the list of new Beacons
+  val newBeacons = mutableListOf<Beacon>()
+
+  // Get the nearby points of interest
+  val nearbyPOIs = getNearbyPOIs(context, location, radius)
+
+  // Place new beacons at the nearby points of interest
+  for (poi in nearbyPOIs) {
+    // Check if POI is far enough from existing beacons
+    if (nearbyBeacons.any() { beacon ->
+      beacon.location.distanceBetween(poi) > MIN_BEACON_DISTANCE
+    }) {
+      newBeacons.add(Beacon("", poi))
+    }
+  }
+
+  return newBeacons
+}
+
+/**
+ * Get the nearby points of interest (POIs) using the Places API.
+ *
+ * @param context The context of the application.
+ * @param location The current location of the user.
+ * @param radius The radius in which to search for POIs.
+ * @return A list of nearby POIs.
+ */
+fun getNearbyPOIs(context: Context, location: Location, radius: Double): List<Location> {
+  // Initialize the list of nearby POIs
+  val nearbyPOIs = mutableListOf<Location>()
+
+  // Initialize the Places API
+  Places.initialize(context, MAPS_API_KEY)
+
+  // Create a new Places client instance
+  val placesClient = Places.createClient(context)
+
+  // Define the fields to request
+  val placeFields = listOf(Place.Field.ID, Place.Field.NAME, Place.Field.LAT_LNG)
+
+  // Create a new FindCurrentPlaceRequest
+  val request = FindCurrentPlaceRequest.newInstance(placeFields)
+
+  // Check if the app has the required permissions
+  if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) ==
+      PackageManager.PERMISSION_GRANTED) {
+    // Use the Places API to find nearby places
+    placesClient
+        .findCurrentPlace(request)
+        .addOnSuccessListener { response ->
+          for (placeLikelihood in response.placeLikelihoods) {
+            val place = placeLikelihood.place
+            // Conversion to make computing distances easier
+            val placeLoc = Location(place.latLng.latitude, place.latLng.longitude, place.name)
+
+            if (location.distanceBetween(placeLoc) <=
+                radius && // Check if the place is within the radius
+                place.reviews.size >= MIN_REVIEWS && // Check if the place has enough reviews
+                place.placeTypes.any() {
+                  types.contains(it)
+                } // Check if the place is of a certain type
+            ) {
+              nearbyPOIs.add(placeLoc)
+              Log.d("PlacesApi", "Place found: ${place.name} at ${place.latLng}")
+            }
+          }
+        }
+        .addOnFailureListener { exception ->
+          if (exception is ApiException) {
+            Log.d("PlacesApi", "Place not found: ${exception.statusCode}")
```

Comment:
In this case, Log.e is better i think for the errors

Reply:
but its just a detail lol

Reply:
You're right, I'll edit it